src/client.ts

```diff
@@ -433,17 +433,39 @@ export class InnerConnection {
     return this._isClosed;
   }
 
+  logConnectionError(...args: any): void {
+    if (!this.config.logging) {
+      return;
+    }
+    if (
+      this.config.inProject &&
+      !this.config.fromProject &&
+      !this.config.fromEnv
```

Comment:
This looks like doesn't ensure that there actually were any parameters to `.connect()`. And I think it may be confusing to see `Consider calling "edgedb.connect()" without arguments.` when there aren't any. I would desperately try to find actual arguments created in some layer of abstractions, while in fact it maybe instance just isn't running.

Reply:
Let's fix this in later iterations. I agree this can be confusing, but I'd like @jaclarke to finalize his connection testing/rework before I add code there.